src/protagonist/Portal/Pages/Queue/Index.cshtml.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Threading.Tasks;
+using API.Client;
+using DLCS.Core.Enum;
+using DLCS.HydraModel;
+using Hydra.Collections;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Portal.ViewComponents;
+
+namespace Portal.Pages.Queue;
+
+public class IndexModel : PageModel
+{
+    private readonly IDlcsClient dlcsClient;
+    private const string ActiveQueueName = "active";
+    private const string RecentQueueName = "recent";
+    public HydraCollection<Batch> Batches { get; set; }
+    public CustomerQueue Queue { get; set; }
+    public BatchQueueType QueueType { get; set; }
+    public PagerValues? PagerValues { get; private set; }
+    
+    public IndexModel(IDlcsClient dlcsClient)
+    {
+        this.dlcsClient = dlcsClient;
+    }
+
+    public async Task OnGetAsync(
+        [FromRoute] string queueType, 
+        [FromQuery] int page = 1, 
+        [FromQuery] int pageSize = PagerViewComponent.DefaultPageSize)
+    {
+        Queue = await dlcsClient.GetQueue();
```

Comment:
Values from `Queue` is only used `@if (Model.QueueType == IndexModel.BatchQueueType.Active)` so only needs to be called if it's active queue.

Reply:
Good catch. Will fix.